src/WMBot/System/Channel.cs

```diff
@@ -108,6 +108,15 @@ public override string TargetName
         public Security SystemUsers = null;
         private bool IsRemoved = false;
 
+        /// <summary>
+        /// Is it restricted to join this channel due to it being invite only
```

Comment:
The reason why channel is restricted is not because it's invite only, but because bot operators don't want anyone to add it there ;)

Reply:
Right now bot sends "{channel_name} was restricted due to it being invite only. Please contact an admin to have it readded" to everybody who tries to add that bot to one of restricted channels. So I initially thought that this functionality will be used only for case of invite only channels.

But if the range of usecases is wider, then we maybe should add a way to specify the reason, or change message to "It was restricted to add bot to {channel_name}. Please contact an admin to have it readded".

What do you think about this?